check_haproxy.rb

```diff
@@ -85,16 +91,28 @@
   exit UNKNOWN
 end
 
+
+tries = 2
+
 begin
   f = open(options.url, :http_basic_authentication => [options.user, options.password])
 rescue OpenURI::HTTPError => e
   puts "ERROR: #{e.message}"
-  exit UNKNOWN
+  exit CRITICAL
 rescue Errno::ECONNREFUSED => e
   puts "ERROR: #{e.message}"
-  exit UNKNOWN
+  exit CRITICAL
+rescue Exception => e
+  if e.message =~ /redirection forbidden/
```

Comment:
I agree that open-uri follows redirect but the issue is that I want to redirect from http:// to https://. There is even gem for this "bug" http://stackoverflow.com/a/27411667/1751337

There is bug report: https://bugs.ruby-lang.org/issues/859